docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,128 @@
+associated with the ECMAScript ecosystem for a number of reasons:
+
+1. Entropic is intended to allow for federated package management for more than just the ECMAScript ecosystem
+1. The `package.json` name specifier does not allow for the namespacing that entropic provides
+1. There are recognized deficiencies with using JSON as human-readable metadata format
+
+## Filename
+
+The filename of the manifest **MUST** be `Package.toml` and it **MUST** live in
+the root directory of the package's filesystem. The root of the package is defined
+as the closest parent directory that contains the manifest file, starting with
+the current directory.
+
+On case-insensitive filesystems, it **MAY** be presented as `package.toml` but care
+should be taken to preserve case when possible.
+
+## Format
+
+The format of this file is **REQUIRED** to be in [TOML](https://github.com/toml-lang/toml)
+version 0.5.0 or later as defined by the project.
+
+### Top-Level Data
+
+The manifest **MUST** contain the following top-level keys:
+
+- `"name"` - the canonical name of the package, as a string, including it's namespace. e.g. `"toddself@static-pkg.dev/packge2toml"`
+- `"version"` - a [SEMVER](https://semver.org/) designation conforming to version 2.0.0 of the semantic versioning specification. Due to how TOML parses numbers, this must be represented as a string. e.g. `"1.0.4"`
+
+The manifest **MAY** contain the following fields:
+
+- `"entry"` - the filename and package-relative path, as a string, to the main entry point of the package. The root for the relative-path is the directory which contains the `Package.toml` file. e.g `"./src/index.js"`
+- `"type"` - if set to `"module"` this will output a [`package.json` with this included](https://nodejs.org/api/esm.html#esm_code_package_json_code_code_type_code_field), and will signal to the node interpreter to run this code an ES6 Module.
+- `"license"` - a valid [SPDX license](https://spdx.org/licenses/) for your code. If a license is not provided then the code is considered to be `UNLICENSED`.
+- `"description"` - a short description of your package
+- `"homepage"` - a URL for your project's homepage
+- `"author"` - who wrote this package
+- `"repository"` - where does this code live
+
+These fields **MAY** or **MAY NOT** be validated by CLI tools or the server-side
+software, depending on the implementation.
+
+Software implementing the entropic federation protocols and `Package.toml` handling **MAY** complain about any of these optional fields as missing (`license` being an obvious candidate).
+
+### Subsections
+
+The manifest file **MAY** contain zero or more subsections as defined below.
+
+#### Dependency Lists
+
+Dependencies lists contain all the resources that are required for
+this package, in various scenarios. The dependencies **MUST** be declared as `"key" = "value"`
+pairs where the `key` is the package's canonical name (as from the top-level section
```

Comment:
We have a unique opportunity here to move away from this restriction from the original `package.json` and add optional support for objects with extra metadata. This has long been a thorn on our side, and I'd love to see the injustice corrected.

This could _possibly_ translate to the disappearance of `[devDependencies]` and such, or make those categories optional in favor of, e.g. `"toddself@static-pkg.dev/package2toml" = { version: "1.0.4", dev: true, optional: true }`.

Reply:
i'm so into in this. leaving the future open for a package manager being able to handle more dependency environments than dev & prod. (for instance, i'd love to be able to specify which dependencies are the `test` dependencies)

Reply:
yes! your insights are super helpful! thank you

Reply:
YES this. I'd also like to reserve the ability for applications to "open" a dependency to patch it, a la Cargo:

```toml
[dependencies."chris@registry.neversaw.us/foo".dependencies."lodash"]
patch = "chris@registry.neversaw.us/underscore"
```